### PR TITLE
feat: add style in component button

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

#1307 

### Description and Example

Add support to style id in component button

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: flutter/beagle/lib/model/beagle_button_style.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ */
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class BeagleButtonStyle {
+  BeagleButtonStyle(
+      {this.androidButtonStyle, this.iosButtonStyle, this.buttonTextStyle});
+
+  /// Customizes this button's appearance to platform Android.
```
Comment (matheusribeirozup) [inline flutter/beagle/lib/model/beagle_button_style.dart:24]:
```suggestion
  /// Customizes the appearance of this button for the Android platform.
```

File: flutter/beagle/lib/model/beagle_button_style.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  /// Null by default.
+  final ButtonStyle androidButtonStyle;
+
+  /// Customizes this button's appearance to platform ios.
```
Comment (matheusribeirozup) [inline flutter/beagle/lib/model/beagle_button_style.dart:28]:
```suggestion
  /// Customizes the appearance of this button for the iOS platform.
```

File: flutter/beagle_components/test/beagle_button_test.dart [hunk 3]
```diff
@@ -132,5 +138,121 @@ void main() {
...
+            buttonOne.iosButtonStyle.pressedOpacity);
+        expect(buttonCreated.color, buttonOne.iosButtonStyle.color);
+
+        expect(textCreated.style, buttonOne.buttonTextStyle);
+
+        debugDefaultTargetPlatformOverride = null;
+      });
+    });
+
+    group('When not set style and platform is Android', () {
```
Comment (matheusribeirozup) [inline flutter/beagle_components/test/beagle_button_test.dart:208]: This group is duplicated

File: flutter/beagle_components/test/beagle_button_test.dart [hunk 4]
```diff
@@ -132,5 +138,121 @@ void main() {
...
+            buttonOne.iosButtonStyle.pressedOpacity);
+        expect(buttonCreated.color, buttonOne.iosButtonStyle.color);
+
+        expect(textCreated.style, buttonOne.buttonTextStyle);
+
+        debugDefaultTargetPlatformOverride = null;
+      });
+    });
+
+    group('When set style and platform is iOS', () {
```
Comment (matheusribeirozup) [inline flutter/beagle_components/test/beagle_button_test.dart:175]: This group is duplicated

File: flutter/beagle_components/test/beagle_button_test.dart [hunk 5]
```diff
@@ -132,5 +138,121 @@ void main() {
...
+        expect(buttonFinder, findsOneWidget);
+        expect(buttonCreated.style, null);
+
+        expect(textCreated.style, null);
+
+        debugDefaultTargetPlatformOverride = null;
+      });
+    });
+
+    group('When not set style and platform is Android', () {
```
Comment (matheusribeirozup) [inline flutter/beagle_components/test/beagle_button_test.dart:233]: This group is also duplicated

File: flutter/beagle_components/test/beagle_button_test.dart (hunk 4 above)
Comment (uziasferreirazup) [reply flutter/beagle_components/test/beagle_button_test.dart]: done

File: flutter/beagle_components/test/beagle_button_test.dart (hunk 5 above)
Comment (uziasferreirazup) [reply flutter/beagle_components/test/beagle_button_test.dart]: done

**Review by matheusribeirozup — APPROVED** [review]

**Review by paulomeurerzup — APPROVED** [review]

**Pull request merged**